perun-core/src/main/java/cz/metacentrum/perun/core/bl/FacilitiesManagerBl.java

```diff
@@ -272,7 +272,7 @@
 	 * @return list of users
 	 * @throws InternalErrorException
 	 */
-	List<User> getAllowedUsersNotExpired(PerunSession sess, Facility facility, Vo specificVo, Service specificService) throws InternalErrorException;
+	List<User> getAllowedUsersNotExpiredInGroup(PerunSession sess, Facility facility, Vo specificVo, Service specificService) throws InternalErrorException;
```

Comment:
Well, we have only two expirations, in the Vo and the Group. If we state only "NotExpired", I expect it to be VO expiration, since its historically more used, but its not the case of this method.
We do not expire on the Resource or Facility. I'm not aware we are using this term anywhere else. Also Resource is not between method params either.

The correct version seems to be rather long, since you have to specify, that by expired you mean group expiration and that you return all users, which are connected/assigned through at least one group, in which they are not expired and that the member as a whole is allowed.

How about ``getAllowedUsersWithoutExpiredIn(Assigned)Groups`` or ``getAllowedUsersOnlyForValidGroups`` or ``getAllowedUsers(Throught/Of)NotExpiredGroups`` ?

Generally, I don't think that we can always match all conditions to the method name for every method. When in doubt, user should read the javadoc, which is more explanatory.

Reply:
We should probably create a new way how to say:
- allowed in VO
- valid in group
Something like we said in user life cycle - Active. For now we can let it this way, but in the future, we should consider something like getActiveUsers instead of this.

Reply:
You are definitely right on that, but we are not there yet with the user life-cycle :-(